### PR TITLE
Add SimpleDataPack for saving data.

### PR DESCRIPTION
**Pull request by BlueCocoa**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/Llama-Group/Project-Llama/pull/161" target="_blank">ReviewNinja

File: llama_cpp/example/simple_data_pack/main.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    
+    pack.push_pure_data("cstring", (void *)"Hello\0", 6);
+    
+    for (int i = 0; i < 4; i++) {
+        SimpleObject object;
+        object.data = i * 233;
+        pack.push(SimpleDataPack::data_t::SIMPLE_OBJECT, "object", &object);
+    }
+    
+    pack.saveTo("233.dat");
```
Comment (al1enSuu) [inline llama_cpp/example/simple_data_pack/main.cpp]: 233?

File: llama_cpp/example/simple_data_pack/main.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
+//
+//  main.cpp
+//  SimpleDataPack
+//
+//  Created by BlueCocoa on 2016/9/30.
+//  Copyright © 2016 BlueCocoa. All rights reserved.
+//
```
Comment (0xsuu) [inline llama_cpp/example/simple_data_pack/main.cpp:7]: Wrong copyright.

File: llama_cpp/src/Utility/SimpleDataPack.h [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        BIT_64          = 0b00001000,
+        BIT_128         = 0b00010000,
+        
+        // Object with basic data type
+        SIMPLE_OBJECT   = 0b00100000,
+        
+        // like void *
+        PURE_DATA       = 0b01000000,
+        
+        // rest values are reserved
```
Comment (0xsuu) [inline llama_cpp/src/Utility/SimpleDataPack.h:36]: Sentence!

File: llama_cpp/src/Utility/SimpleDataPack.h [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        BIT_8           = 0b00000001,
+        BIT_16          = 0b00000010,
+        BIT_32          = 0b00000100,
+        BIT_64          = 0b00001000,
+        BIT_128         = 0b00010000,
+        
+        // Object with basic data type
+        SIMPLE_OBJECT   = 0b00100000,
+        
+        // like void *
```
Comment (0xsuu) [inline llama_cpp/src/Utility/SimpleDataPack.h:33]: Sentence!

File: llama_cpp/src/Utility/SimpleDataPack.h [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+class SimpleDataPack {
+ public:
+    typedef enum data_t : int16_t {
+        BIT_8           = 0b00000001,
+        BIT_16          = 0b00000010,
+        BIT_32          = 0b00000100,
+        BIT_64          = 0b00001000,
+        BIT_128         = 0b00010000,
+        
+        // Object with basic data type
```
Comment (0xsuu) [inline llama_cpp/src/Utility/SimpleDataPack.h:30]: Sentence! Missing full stop.

File: llama_cpp/src/Utility/SimpleDataPack.h [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        if (item) {
+            memset(reinterpret_cast<void *>(item), 0, sizeof(struct savable_list_t));
+            item->data_type = (data_t)(true << 15);
+            item->data_type |= data_t::PURE_DATA;
+            item->id = std::make_shared<std::string>(id);
+            
+            item->copied = malloc(length);
+            if (item->copied) {
+                memset(item->copied, 0, length);
+                memcpy(item->copied, (const void *)data, length);
```
Comment (0xsuu) [inline llama_cpp/src/Utility/SimpleDataPack.h:95]: Shouldn't use c style type cast.

**Comment by BlueCocoa** [comment]
No more interested in this.

**Pull request closed without merging**